src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -295,8 +295,11 @@ static void uct_cuda_copy_sync_memops(uct_cuda_copy_md_t *md,
     CUdriverProcAddressQueryResult sym_status;
     CUresult cu_err;
     ucs_status_t status;
+    khiter_t iter;
+    ucs_kh_put_t ret;
 
-    if (md->sync_memops_set) {
+    iter = kh_get(cuda_ctx_set, &md->sync_memops, (khint64_t)ctx);
```

Comment:
why do we need a set? Is it an optimization?
do we need both - map and set? Maybe it is possible to maintain a single map?
also seems to be related to #9645

Reply:
`uct_cuda_retained_primary_ctxs` hash map associates the device with its retained primary device context. Device handle is a key, context handle is a value.
`uct_cuda_copy_md_t::sync_memops` is a hash set of context handles with the `CU_CTX_SYNC_MEMOPS` flag set.

Reply:
Before we assumed that only one CUDA context handle is used. Thus, we only used a boolean value to store information that the flag is set.